### PR TITLE
Feature/1240 enhance some misc functions

### LINKED ISSUE #939: NumericWaveToList: Add support for 2D waves
This works with wfprintf as well, see

```igorpro
Function Example1()
	Make/O/N=(10, 2) wave0
	wave0[][0] = sin(p*pi/10)
	wave0[][1] = 1

	wfprintf 1, "", wave0
	
	print "##############"

	wfprintf 1, "%g %g\r", wave0
End
```

### PR DESCRIPTION
**Pull request by t-b**
Preparations for #1238.

@MichaelHuth Please review and can you also provide a fix for `TWTLRoundTrips:4`? IMHO the error is in ListToTextWaveMD.

Close #939.

### CODE REVIEW

File: Packages/MIES/MIES_Utilities.ipf [hunk 1]
```diff
@@ -3144,10 +3144,12 @@ End
 ///
 /// @param wv     numeric wave
 /// @param sep    separator
+/// @param colSep [optional, default = `,`] separator for column entries
 /// @param format [optional, defaults to `%g`] sprintf conversion specifier
-threadsafe Function/S NumericWaveToList(WAVE/Z wv, string sep, [string format])
-
+threadsafe Function/S NumericWaveToList(WAVE/Z wv, string sep, [string format, string colSep])
```
Comment (timjarsky) [inline Packages/MIES/MIES_Utilities.ipf:3228]: @t-b are in-line return parameters (e.g., ```Function/S [string list] NumericWaveToList(WAVE/Z wv, string sep, [string format, string colSep])``` ) only used when there is >1 return param?

**Review by timjarsky — DISMISSED** [review]

File: Packages/MIES/MIES_Utilities.ipf (hunk 1 above)
Comment (t-b) [reply Packages/MIES/MIES_Utilities.ipf:3228]: Jeep. This is done as using the single-return signature is nicer at the call site.

In the following example three different use cases are tried with classic and MRS (multiple return value syntax) functions. Only the classic one can be used in all three cases, MRS fails in two.

```igorpro
Function/S FuncClassic()

	return ""
End

Function [string str] FuncMRS()

	return [""]
End

Function Consume(string str)

End

Function Test()
	string str1, str2

	// ignoring return value works
	FuncClassic()

	// using it
	str1 = FuncClassic()

	// pass onto caller
	Consume(FuncClassic())

	// does not compile
	// FuncMRS()
	
	[str2] = FuncMRS()
	
	// does not compile
	// Consume(FuncMRS())
End

```

File: Packages/MIES/MIES_Utilities.ipf [hunk 2]
```diff
@@ -3116,6 +3116,26 @@ threadsafe Function/WAVE ListToTextWaveMD(list, dims, [rowSep, colSep, laySep, c
...
+threadsafe Function/S ReplicateString(string str, variable numTotalCopies)
+
+	variable i
+	string list = ""
+
+	if(!IsFinite(numTotalCopies) || numTotalCopies <= 0)
+		return str
+	endif
+
+	for(i = 0; i < numTotalCopies; i += 1)
```
Comment (MichaelHuth) [inline Packages/MIES/MIES_Utilities.ipf:3209]: Just a minor thought:
I have not benched it, though I would have used a two liner with
PadString + ReplaceString here.

File: Packages/MIES/MIES_Utilities.ipf [hunk 3]
```diff
@@ -3156,14 +3158,28 @@ threadsafe Function/S NumericWaveToList(WAVE/Z wv, string sep, [string format])
 		format = "%g"
 	endif
 
+	if(ParamIsDefault(colSep))
+		colSep = ","
```
Comment (MichaelHuth) [inline Packages/MIES/MIES_Utilities.ipf:3241]: Would it make sense to have some default separator constants defined globally that are used preferrentially?
If yes, lets make an issue for that.

**Comment by t-b** [comment]
Just for the record. The error was in my TextWaveToList changes and not in ListToTextWaveMD.

**Review by MichaelHuth — APPROVED** [review]
See comments, the rest looks good to me.

File: Packages/MIES/MIES_Utilities.ipf (hunk 2 above)
Comment (t-b) [reply Packages/MIES/MIES_Utilities.ipf:3209]: Might be worth a thought. But IP8 is also on its way out so I'd leave it as it is.

File: Packages/MIES/MIES_Utilities.ipf (hunk 3 above)
Comment (t-b) [reply Packages/MIES/MIES_Utilities.ipf:3241]: Good point. An issue would be good.

**Pull request merged**